src/main/java/com/rackspace/salus/monitor_management/services/MonitorManagement.java

```diff
@@ -2276,4 +2278,11 @@ public RenderedMonitorTemplate renderMonitorTemplate(UUID monitorId, String reso
     }
     return renderedMonitorTemplate;
   }
+
+  public ResourceDTO findResourceByTenantIdAndResourceId(String tenantId, String resourceId) {
+    Resource resource = resourceRepository.findByTenantIdAndResourceId(tenantId, resourceId)
+        .orElse(null);
+
+    return resource == null ? null : new ResourceDTO(resource, null);
```

Comment:
With my clarification, I meant:
```suggestion
    return resourceRepository.findByTenantIdAndResourceId(tenantId, resourceId)
        .map(resource -> new ResourceDTO(resource,null))
        .orElse(null);
```